File: apps/api/app/schemas/leaders.py
"""Schemas for leaders endpoints."""

from pydantic import BaseModel

from app.schemas.common import NationInfo


class CareerTotalsPlayer(BaseModel):
    """Player career totals information."""

    player_id: int
    player_name: str
    nation: NationInfo | None
    total_goal_value: float
    goal_value_avg: float
    total_goals: int
    total_matches: int


class CareerTotalsResponse(BaseModel):
    """Response for career totals leaders."""

    top_goal_value: list[CareerTotalsPlayer]


class BySeasonPlayer(BaseModel):
    """Player by season information."""

    player_id: int
    player_name: str
    clubs: str
    total_goal_value: float
    goal_value_avg: float
    total_goals: int
    total_matches: int


class BySeasonResponse(BaseModel):
    """Response for by season leaders."""

    top_goal_value: list[BySeasonPlayer]

File: apps/web/lib/types/leaders.ts
export interface NationInfo {
  id: number | null
  name: string | null
  country_code: string | null
}

export interface CareerTotalsPlayer {
  player_id: number
  player_name: string
  nation: NationInfo | null
  total_goal_value: number
  goal_value_avg: number
  total_goals: number
  total_matches: number
}

export interface CareerTotalsResponse {
  top_goal_value: CareerTotalsPlayer[]
}

export interface BySeasonPlayer {
  player_id: number
  player_name: string
  clubs: string
  total_goal_value: number
  goal_value_avg: number
  total_goals: number
  total_matches: number
}

export interface BySeasonResponse {
  top_goal_value: BySeasonPlayer[]
}

export interface AllSeasonsPlayer {
  player_id: number
  player_name: string
  season_id: number
  season_display_name: string
  clubs: string
  total_goal_value: number
  goal_value_avg: number
  total_goals: number
  total_matches: number
}

export interface AllSeasonsResponse {
  top_goal_value: AllSeasonsPlayer[]
}
